esx_service/utils/auth_data.py

```diff
@@ -115,19 +117,44 @@ def remove_vms(self, conn, vms):
 
         return None
 
-    def set_name(self, conn, name):
+    def set_name(self, conn, name, new_name):
         """ Set name column in tenant table for this tenant. """
+        logging.debug("set_name: name=%s, new_name=%s", name, new_name)
         tenant_id = self.id
         try:
             conn.execute(
                 "UPDATE tenants SET name = ? WHERE id = ?",
-                (name, tenant_id)
+                (new_name, tenant_id)
             )
             conn.commit()
         except sqlite3.Error as e:
             logging.error("Error %s when updating tenants table with tenant_id"
                           "tenant_id %s", e, tenant_id)
             return str(e)
+        
+        # rename in the DB succeeds
+        # unlink the old symbol link /vmfs/volumes/datastore_name/tenant_name
+        # create the new symbol link /vmfs/volumes/datastore_name/new_tenant_name 
+        # which point to path /vmfs/volumes/datastore_name/tenant_uuid
+        lock = threadutils.get_lock()
+        for (datastore, url_name, path) in vmdk_utils.get_datastores():
+            dock_vol_path = os.path.join("/vmfs/volumes", datastore, vmdk_ops.DOCK_VOLS_DIR)
+            tenant_path = os.path.join(dock_vol_path, tenant_id)
+            logging.debug("set_name: try to update the symlink to path %s", tenant_path)
+            
+            if os.path.isdir(tenant_path):
+                lock.acquire() 
```

Comment:
What exactly this lock is protecting? 

Is there anything that synchronizes  `docker volume create` and `admin_cli tenant update` happening at the same time?